Library/src/Multiplayer/Components/HotspotSpaceComponent.cpp

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2023 Magnopus LLC
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "CSP/Multiplayer/Components/HotspotSpaceComponent.h"
+#include "CSP/Multiplayer/SpaceEntity.h"
+#include "Debug/Logging.h"
+#include "Memory/Memory.h"
+#include "Multiplayer/Script/ComponentBinding/HotspotSpaceComponentScriptInterface.h"
+
+
+namespace csp::multiplayer
+{
+
+HotspotSpaceComponent::HotspotSpaceComponent(SpaceEntity* Parent) : ComponentBase(ComponentType::Hotspot, Parent), UniqueComponentId("")
+{
+	Properties[static_cast<uint32_t>(HotspotPropertyKeys::Position)]			 = csp::common::Vector3::Zero();
+	Properties[static_cast<uint32_t>(HotspotPropertyKeys::Rotation)]			 = csp::common::Vector4 {0, 0, 0, 1};
+	Properties[static_cast<uint32_t>(HotspotPropertyKeys::Name)]				 = "";
+	Properties[static_cast<uint32_t>(HotspotPropertyKeys::HotspotType)]			 = static_cast<int64_t>(csp::multiplayer::HotspotType::TeleportHotspot);
+    Properties[static_cast<uint32_t>(HotspotPropertyKeys::IsVisible)]			 = true;
+	Properties[static_cast<uint32_t>(HotspotPropertyKeys::IsARVisible)]		     = true;
+
+	SetScriptInterface(CSP_NEW HotspotSpaceComponentScriptInterface(this));
+}
+
+const csp::common::String& HotspotSpaceComponent::GetName() const
+{
+	return GetStringProperty(static_cast<uint32_t>(HotspotPropertyKeys::Name));
+}
+
+void HotspotSpaceComponent::SetName(const csp::common::String& Value)
+{
+	SetProperty(static_cast<uint32_t>(HotspotPropertyKeys::Name), Value);
+}
+
+HotspotType HotspotSpaceComponent::GetHotspotType() const
+{
+	return static_cast<HotspotType>(GetIntegerProperty(static_cast<uint32_t>(HotspotPropertyKeys::HotspotType)));
+}
+
+void HotspotSpaceComponent::SetHotspotType(HotspotType Value)
+{
+	SetProperty(static_cast<uint32_t>(HotspotPropertyKeys::HotspotType), static_cast<int64_t>(Value));
+}
+
+const csp::common::String& HotspotSpaceComponent::GetUniqueComponentId()
```

Comment:
Why is UniqueComponentId a property of the class? Is it worth removing this and making the function const?